src/controller/python/chip/bdx/bdx.cpp

```diff
@@ -0,0 +1,243 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <vector>
+
+#include <protocols/bdx/BdxMessages.h>
+
+#include <controller/CHIPDeviceControllerFactory.h>
+#include <controller/CHIPDeviceControllerSystemState.h>
+#include <controller/python/chip/bdx/bdx-transfer.h>
+#include <controller/python/chip/bdx/test-bdx-transfer-server.h>
+#include <controller/python/chip/native/PyChipError.h>
+
+using PyObject = void *;
+
+namespace chip {
+namespace python {
+
+// The Python callbacks to call when certain events happen.
+using OnTransferObtainedCallback  = void (*)(PyObject context, PyChipError result, void * bdxTransfer,
+                                            bdx::TransferControlFlags transferControlFlags, uint16_t maxBlockSize,
+                                            uint64_t startOffset, uint64_t length, const uint8_t * fileDesignator,
+                                            uint16_t fileDesignatorLength, const uint8_t * metadata, size_t metadataLength);
+using OnDataReceivedCallback      = void (*)(PyObject context, const uint8_t * dataBuffer, size_t bufferLength);
+using OnTransferCompletedCallback = void (*)(PyObject context, PyChipError result);
+
+OnTransferObtainedCallback gOnTransferObtainedCallback   = nullptr;
+OnDataReceivedCallback gOnDataReceivedCallback           = nullptr;
+OnTransferCompletedCallback gOnTransferCompletedCallback = nullptr;
+
+// The information for a single transfer.
+struct TransferData
```

Comment:
The name of this struct is a bit odd because it's not the data for transfer, its the information about the transfer. Can this be TranferInfo or some other such thing